src/main/java/org/gridsuite/study/server/service/ReportService.java

```diff
@@ -50,6 +58,126 @@ public ReportService(
+            UUID downingDefNodeUuid = uppingNeededDefNodeUuids.removeLast();
+            ReportingInfos info = definingNodesInfosByUuid.get(downingDefNodeUuid);
+
+            ReporterModel mayReporter = seenBuildNodeToRootReporter.get(downingDefNodeUuid);
+            boolean wantSubs = mayReporter == null;
+            if (wantSubs) {
+                UUID buildNodeUuid = seenDefToBuildNodeUuuids.get(downingDefNodeUuid);
+                mayReporter = seenBuildNodeToRootReporter.get(buildNodeUuid);
+            }
+
+            ReporterModel newReporter = filterAndAdaptReporter(groupToDefiningNode, downingDefNodeUuid, info,
+                mayReporter, wantSubs);
+
+            res.add(newReporter);
+        }
+
+        return res;
+    }
+
+    private static ReporterModel filterAndAdaptReporter(Map<UUID, UUID> groupToDefiningNode, UUID downingDefNodeUuid,
+            ReportingInfos info, ReporterModel receivedReporter, boolean wantsSubs) {
+
+        ReporterModel newReporter = new ReporterModel(receivedReporter.getDefaultName(),
+            info.getDefiningNodeName(), receivedReporter.getTaskValues());
+
+        for (ReporterModel subReporter : receivedReporter.getSubReporters()) {
+            UUID groupUuid = getGroupUuidFromReporter(subReporter);
+
+            boolean avoids = groupUuid == null ? wantsSubs : !Objects.equals(downingDefNodeUuid, groupToDefiningNode.get(groupUuid));
+
+            if (avoids) {
+                continue;
+            }
+
+            ReporterModel newSub = new ReporterModel(subReporter.getDefaultName(), subReporter.getDefaultName(), subReporter.getTaskValues());
+            newReporter.addSubReporter(newSub);
+            subReporter.getReports().forEach(newSub::report);
+            subReporter.getSubReporters().forEach(newSub::addSubReporter);
+        }
+
+        return newReporter;
+    }
+
+    private void fillMapsOfSeenNodes(List<UUID> neededDefiningNodeUuids,
+            Map<UUID, UUID> groupToDefiningNode,
+            Map<UUID, ReportingInfos> definingNodesInfosByUuid,
+            Map<UUID, UUID> seenDefToBuildNodeUuuids,
+            Map<UUID, ReporterModel> seenBuildNodeToRootReporter) {
+
+        for (UUID buildNodeUuid : neededDefiningNodeUuids) {
+            if (seenDefToBuildNodeUuuids.containsKey(buildNodeUuid)) {
+                continue;
+            }
+
+            ReportingInfos reportingInfo = definingNodesInfosByUuid.get(buildNodeUuid);
+            UUID reportUuid = reportingInfo.getReportUuid();
+
+            // getReport asks to silently return empty reportmodel if not found.
+            ReporterModel reporter = this.getReport(reportUuid, "useless roundtrip string");
+            if (reporter == null) {
+                continue;
+            }
+
+            seenBuildNodeToRootReporter.put(buildNodeUuid, reporter);
+
+            for (ReporterModel subReport : reporter.getSubReporters()) {
+                UUID groupUuid = getGroupUuidFromReporter(subReport);
```

Comment:
How can you extract a groupUuid from subReport task key ?
In modification-server, taskey is never an UUID ...

Reply:
Thanks for noticing that ! I push a wrong branch. One can now check https://github.com/gridsuite/network-modification-server/pull/129